llvm-general/src/LLVM/General/OrcJIT.hs

```diff
@@ -12,7 +12,12 @@ module LLVM.General.OrcJIT (
     mangleSymbol,
     withIRCompileLayer,
     withModuleSet,
-    withObjectLinkingLayer
+    withObjectLinkingLayer,
+    addModuleSet,
+    createObjectLinkingLayer,
```

Comment:
Yes! Providing `with` wrappings is very important, I certainly did not want to suggest removing them. 

At the risk of stating the obvious, a limitation of the current `with` arrangement is that it fixes the outermost type constructor of the user's code to `IO`. This makes it challenging to use `llvm-general` in a larger program: `llvm-general` is setup somewhat like a classic OO-style framework that insists on being the outer wrapping of whatever else you wish to do. This could probably be mitigated within the current design, but exposing the pieces necessary for downstream libraries to build their own lifecycle management is essential.